p2p/test/muxer-negotiation/muxer_negotiation_test.go

```diff
@@ -0,0 +1,199 @@
+package muxernegotiation
+
+import (
+	"bufio"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io"
+	"log"
+	"testing"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/p2p/muxer/mplex"
+	"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	"github.com/stretchr/testify/require"
+
+	/*
+		"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	*/
+
+	golog "github.com/ipfs/go-log/v2"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const testData = "Muxer Test Data\n"
+
+type muxerTest struct {
+	svrMuxers     []string
+	svrTrans      []network.Multiplexer
+	cliMuxers     []string
+	cliTrans      []network.Multiplexer
+	expectedMuxer string
+}
+
+func TestMuxerNegotiatin(t *testing.T) {
+	secTypes := []string{Tls, Noise}
+	testCases := []muxerTest{
+		{svrMuxers: []string{"/mplex/6.7.0"},
+			svrTrans:      []network.Multiplexer{mplex.DefaultTransport},
+			cliMuxers:     []string{"/yamux/1.0.0"},
```

Comment:
It's probably nicer to introduce a new struct to group the name and muxer together.